pkg/apis/kops/cluster.go

```diff
@@ -233,6 +233,33 @@ type ClusterSpec struct {
 
 	// Networking configuration
 	Networking *NetworkingSpec `json:"networking,omitempty"`
+
+	// API field controls how the API is exposed outside the cluster
+	API *PublishSpec `json:"api,omitempty"`
+}
+
+type PublishSpec struct {
```

Comment:
If we go with `AccessSpec` we should use `DNSAccessSpec` if we go with `PublishSpec` we should go with `DNSPublishSpec` etc etc..

+1 to `AccessSpec`